validator/keymanager/remote-web3signer/internal/client.go

```diff
@@ -0,0 +1,179 @@
+package internal
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	fieldparams "github.com/prysmaticlabs/prysm/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/crypto/bls"
+	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+)
+
+const (
+	ethApiNamespace = "/api/v1/eth2/sign/"
+	maxTimeout      = 3 * time.Second
+)
+
+type SignRequestJson []byte
+
+// HttpSignerClient defines the interface for interacting with a remote web3signer.
+type HttpSignerClient interface {
+	Sign(ctx context.Context, pubKey string, request SignRequestJson) (bls.Signature, error)
+	GetPublicKeys(ctx context.Context, url string) ([][48]byte, error)
+}
+
+// ApiClient a wrapper object around web3signer APIs. Please refer to the docs from Consensys' web3signer project.
+type ApiClient struct {
+	BaseURL    *url.URL
+	RestClient *http.Client
+}
+
+// NewApiClient method instantiates a new ApiClient object.
+func NewApiClient(baseEndpoint string) (*ApiClient, error) {
+	u, err := url.Parse(baseEndpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid format, unable to parse url")
+	}
+	return &ApiClient{
+		BaseURL: u,
+		RestClient: &http.Client{
+			Timeout: maxTimeout,
```

Comment:
Let's remove this timeout. All requests require a context.Context, which the caller sets the deadline/timeout on the request. There is no need to set a very short 3s timeout when the caller has the ability to do this. Most of these contexts have a deadline set to the end of the slot anyway.